department-of-veterans-affairs/caseflow
# Ensure props change when navigating to a new page from the nav bar

Resolves #15014

### Description
The navigation bar is a [pure component](https://reactjs.org/docs/react-api.html#reactpurecomponent), which will not update if it's props have not changed. 

> React.PureComponent’s shouldComponentUpdate() only shallowly compares the objects. If these contain complex data structures, it may produce false-negatives for deeper differences. Only extend PureComponent when you expect to have simple props and state, or use forceUpdate() when you know deep data structures have changed. Or, consider using immutable objects to facilitate fast comparisons of nested data.
> Furthermore, React.PureComponent’s shouldComponentUpdate() skips prop updates for the whole component subtree. Make sure all the children components are also “pure”.

This causes it's child QueueLoadingScreen and LoadingDataDisplay to not mount again and grab new data when navigating from a judge's assign queue to their normal queue when navigating via the CaseflowNavigationBar.

This change overrides the shallow comparison of props and explicitly checks to see if we are navigating to a new URL.

### Acceptance Criteria
- [x] Judges can navigate from their assign queue to their normal queue via the navigation bar without getting a white screen of death

### Testing Plan
1. Log in as BVAAABSHIRE
1. Navigate to http://localhost:3000/queue/BVAAABSHIRE/assign
1. Click "Caseflow Queue" at the top of the page
1. Notice page now shows loading icon and queue loads correctly
1. Click "Caseflow Queue" at the top of the page
1. Notice no loading icon
